Automattic/jetpack
# Add a feedback admin view switch

## Proposed changes:

This patch adds a 'view switch' which will allow for switching between the classic WP Admin view for the feedback area and the new Jetpack Forms dashboard once it becomes enabled.

The switch is similar to the one already implemented for switching between WP Admin and Calypso views but it is also quite different in that both views are actually part of WP Admin itself.

Choosing an option through the dropdown, will also set that as the preferred option for the future.

![Screenshot 2023-03-02 at 11 31 04](https://user-images.githubusercontent.com/8056203/222404093-ca994c3e-e7cd-4534-b17f-1d0d73642179.png)

### Other information:


## Jetpack product discussion



pcsBup-Pl-p2

## Does this pull request change what data or activity we track or use?



No.

## Testing instructions:






Enable the new dashboard and the switch with these filters:

```
add_filter( 'jetpack_contact_form_use_package', '__return_true' );
add_filter( 'jetpack_forms_dashboard_enable', '__return_true' );
```

- Go to WP Admin and click `Feedback` in the sidebar, you should be redirected to the new dashboard which currently also saves as the default setting.
- There should be a `View` dropdown in the upper right corner of the screen.
- Select `Classic` from the dropdown.
- You should now be on the old WP Admin view.
- If you go to any other WP Admin screen and click back on 'Feedback', it should now automatically show you the WP Admin interface which was your last preferred option.
- Switch back to the `Modern` version.
- Verify that `Modern` is now the default again.
- You should be able to access either view manually using the URLs: `admin.php?page=jetpack-forms` or `edit.php?post_type=feedback`
- Accessing a view through the URL without the `preferred-view` param should not affect your default setting.
